sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/pubsublite/SubscriptionPartitionLoader.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.beam.sdk.io.gcp.pubsublite;
+
+import static org.apache.beam.vendor.guava.v26_0_jre.com.google.common.base.Preconditions.checkArgument;
+
+import com.google.cloud.pubsublite.Partition;
+import com.google.cloud.pubsublite.PartitionLookupUtils;
+import com.google.cloud.pubsublite.SubscriptionPath;
+import com.google.cloud.pubsublite.TopicPath;
+import com.google.common.annotations.VisibleForTesting;
+import java.util.List;
+import java.util.stream.Collectors;
+import java.util.stream.IntStream;
+import org.apache.beam.sdk.transforms.Create;
+import org.apache.beam.sdk.transforms.MapElements;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.SerializableFunction;
+import org.apache.beam.sdk.transforms.Watch;
+import org.apache.beam.sdk.transforms.Watch.Growth.PollFn;
+import org.apache.beam.sdk.transforms.Watch.Growth.PollResult;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PBegin;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.TypeDescriptor;
+import org.apache.beam.vendor.guava.v26_0_jre.com.google.common.collect.ImmutableList;
+import org.joda.time.Duration;
+import org.joda.time.Instant;
+
+class SubscriptionPartitionLoader extends PTransform<PBegin, PCollection<SubscriptionPartition>> {
```

Comment:
`pollDuration` and `terminate` should be configurable from `PubSubLiteIO` with provided default values. What do you think?

Reply:
And for `terminate`, if you decide to use `Watch.Growth.afterTotalOf` as the termination condition, it would be better to have the time duration directly instead of a boolean. What do you think?

Reply:
> should be configurable from PubSubLiteIO

I don't know why? I can't imagine a user would care how often they were polling for an event that is likely to be human triggered. I'd rather defer this until someone asks for it.

> if you decide to use Watch.Growth.afterTotalOf as the termination condition, it would be better to have the time duration directly instead of a boolean

It will never terminate. The boolean only exists for testing purposes, as testing unbounded transforms seems to be an unsolved problem as of yet? If you can think of a better way to test this let me know.

Reply:
> I don't know why? I can't imagine a user would care how often they were polling for an event that is likely to be human triggered. I'd rather defer this until someone asks for it.

If users have a good knowledge on the pattern on partitions growth, they can figure out a reasonable poll duration to be able to detect the new produced partitions as soon as possible, or to avoid doing polling so frequently.

> It will never terminate. The boolean only exists for testing purposes, as testing unbounded transforms seems to be an unsolved problem as of yet? If you can think of a better way to test this let me know.

Hmmm that's a real problem for testing unbounded transform. I don't have a better idea now.

Reply:
> the pattern on partitions growth
This should be O(months) apart. I think the default frequency is fine.